### ISSUE TITLE
Malon Sun's Softlock

### ISSUE THREAD

**Issue by r0bd0g** [TestRunnerSRL/OoT-Randomizer#580]
https://clips.twitch.tv/ConsiderateBreakablePotTakeNRG

This has been around forever.

**Comment by mzxrules** [comment]
proposed fix: #588

Looking at the code, the problem seemed to be caused this bit: 
```
if item.name == 'Suns Song':
    rom.write_byte(locationaddress, next_song_id)
```

Normally when you learn songs, you have a dialog that teaches the song, and a dialog asks you play it. This line set it such that it replaced the dialog that teaches the song with the one that asks to you play it. The problem with this is that after the teach dialog is started Malon swaps to a state where she waits for the teach dialog to complete, and since no teach dialog was initiated she becomes stuck in that state indefinitely. The softlock itself is likely due to Malon setting a Link state flag manually every frame during this state.

**Comment by TestRunnerSRL** [comment]
Fixed now
